app/test/main.test.js

```diff
@@ -0,0 +1,179 @@
+        .end((err, res) => {
+          res.should.have.status(400);
+          done();
+        });
+    });
+  });
+
+  describe('and it does not contain vin', () => {
+    it('then 400 (Bad Request) http code is returned.', (done) => {
+      const correctSmmtConfig = {
+        load: () => ({
+          smmtVincheckUri: fakeRestClient.validSmmtUri,
+          smmtApiKey: fakeVincheck.validSmmtKey,
+        }),
+      };
+
+      const service = proxyquire('../src/main', { './config/loader': correctSmmtConfig, superagent: fakeRestClient });
+
+      chai.request(service.app)
+        .get('/recalls')
+        .query({ make: 'AnyMarque' })
+        .end((err, res) => {
+          res.should.have.status(400);
+          done();
+        });
+    });
+  });
+
+  describe('and it does not contain make', () => {
+    it('then 400 (Bad Request) http code is returned.', (done) => {
+      const correctSmmtConfig = {
+        load: () => ({
+          smmtVincheckUri: fakeRestClient.validSmmtUri,
+          smmtApiKey: fakeVincheck.validSmmtKey,
+        }),
+      };
+
+      const service = proxyquire('../src/main', { './config/loader': correctSmmtConfig, superagent: fakeRestClient });
+
+      chai.request(service.app)
+        .get('/recalls')
+        .query({ vin: 'AnyVin' })
+        .end((err, res) => {
+          res.should.have.status(400);
+          done();
+        });
+    });
+  });
+
+  describe('and it contain vin and make', () => {
+    it('and vehicle has outstanding recall then http 200 and message "Recall Outstanding" it is returned.', (done) => {
+      const correctSmmtConfig = {
+        load: () => ({
+          smmtVincheckUri: fakeRestClient.validSmmtUri,
+          smmtApiKey: fakeVincheck.validSmmtKey,
+        }),
+      };
+
+      const service = proxyquire('../src/main', { './config/loader': correctSmmtConfig, superagent: fakeRestClient });
+
+      chai.request(service.app)
+        .get('/recalls')
+        .query({ make: 'RENAULT', vin: 'AISXXXTEST1239607' })
+        .end((err, res) => {
+          if (err) done(err);
+
+          res.should.have.status(200);
+          res.should.have.header('content-type', 'application/json; charset=utf-8');
+
+          res.body.should.be.a('object');
+          res.body.should.have.property('status_description').eql('Recall Outstanding');
+          res.body.should.have.property('vin_recall_status').eql('BRAKES');
+          res.body.should.have.property('last_update').eql('19022015');
+
+          done();
+        });
+    });
+
+    it('and vehicle has not outstanding recall then http code 200 and message "No Recall Outstanding" it is returned.', (done) => {
+      const correctSmmtConfig = {
+        load: () => ({
+          smmtVincheckUri: fakeRestClient.validSmmtUri,
+          smmtApiKey: fakeVincheck.validSmmtKey,
+        }),
+      };
+
+      const service = proxyquire('../src/main', { './config/loader': correctSmmtConfig, superagent: fakeRestClient });
```

Comment:
perhaps we could extract this as well in all tests this is used in

Reply:
same as above